File: Solution/0011.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int start=0, maxArea=0,end=height.size()-1;
        
        int area;
        while(start < end){
            area = min(height[start], height[end]) * (end - start);
            if(area >= maxArea) maxArea = area;
            
            if(min(height[start], height[end]) == height[start])
                start++;
            else
                end--;
            
        }
        return maxArea;
        
    }
};

File: Solution/0011.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        l, r = 0, n-1

        max_area = 0
        while l < r:
            area = min(height[l], height[r]) * (r-l)
            max_area = max(max_area, area)
            if height[l] <= height[r]:
                l += 1
            else:
                r -= 1
        return max_area